### PR TITLE
[SeaTunnel #804] [patch] Improve the pull_request template

### PR DESCRIPTION
**Pull request by leo65535**
## Purpose of this pull request

We should have a unified pull request name form, like `SeaTunnel #xxx`, it will be useful for us when we want to read the issue.

Add `vcs.xml` will help us jump to issues directly id Intellij IDEA.

image

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] Change does not need document change, or I will submit document change to https://github.com/InterestingLab/seatunnel-docs later

### CODE REVIEW

**Comment by timothy1024** [comment]
cc @CalvinKirs, seems the `Dead Link Checker` is unstable sometime.

```
FILE: ./seatunnel-dist/docker/README.md

  2 links checked.

FILE: ./README_zh_CN.md

  [✖] https://www.qutoutiao.net/

  19 links checked.
  ERROR: 1 dead links found!
  [✖] https://www.qutoutiao.net/ → Status: 0
Error: Process completed with exit code 1.
```

**Comment by leo65535** [comment]
> cc @CalvinKirs, seems the `Dead Link Checker` is unstable sometime.
> 
> ```
> FILE: ./seatunnel-dist/docker/README.md
> 
>   2 links checked.
> 
> FILE: ./README_zh_CN.md
> 
>   [✖] https://www.qutoutiao.net/
> 
>   19 links checked.
>   ERROR: 1 dead links found!
>   [✖] https://www.qutoutiao.net/ → Status: 0
> Error: Process completed with exit code 1.
> ```

thanks for helping check the root cause.

**Comment by leo65535** [comment]
The name form we can discuss, like `[SeaTunnel #xxx]` or `[SeaTunnel-804]` in slack chat.

**Review by CalvinKirs — APPROVED** [review]
LGTM

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -1,19 +1,20 @@
...
-In case of existing issue, reference it using one of the following:
-closes: #ISSUE_ID
-related: #ISSUE_ID
+## Contribution Checklist
 
-Here is some example about how to write a good git commit message:
-http://chris.beams.io/posts/git-commit/
--->
+  - Make sure that the pull request corresponds to a [JIRA issue](https://github.com/InterestingLab/seatunnel/issues).
+
```
Comment (CalvinKirs) [inline .github/PULL_REQUEST_TEMPLATE.md:12]: Should we delete JIRA? we don't use JIRA

Comment (leo65535) [reply .github/PULL_REQUEST_TEMPLATE.md]: how about change `JIRA` to `GITHUB`

**Review by CalvinKirs — COMMENTED** [review]
`.idea/.gitignore ` `.idea/vcs.xml`

don't submit these files

**Comment by leo65535** [comment]
> don't submit these files

`.idea/.gitignore`, good catch, it's my mistake.

`.idea/vcs.xml` this file is the keypoint that help us jump to issuse directly, and it will not cause any problem, refer to
https://github.com/apache/flink/tree/master/.idea

**Comment by CalvinKirs** [comment]
> > don't submit these files
> 
> hi @CalvinKirs, thanks for your review.
> 
> `.idea/.gitignore`, good catch, it's my mistake.
> 
> `.idea/vcs.xml` this file is the keypoint that help us jump to issuse directly, and it will not cause any problem, refer to https://github.com/apache/flink/tree/master/.idea

Thx,got it 👍

**Review by CalvinKirs — APPROVED** [review]
LGTM

**Pull request merged**